### PR TITLE
Modularize build script by splitting it into different functions

### PR DESCRIPTION
**Pull request by sagarkrsd**
- This PR intends to modularize build script by splitting it into various functions so that it become more readable and understandable.

fixes: #57 

Signed-off-by: sagarkrsd <sagar.kumar@openebs.io>

### CODE REVIEW

File: hack/build.sh [hunk 1]
```diff
@@ -3,7 +3,81 @@
...
+    echo ${GIT_COMMIT}
+}
+
+# Delete the old contents
+deleteOldContents(){
+    rm -rf bin/*
+    mkdir -p bin/
+
+    if [ -z "${CTLNAME}" ];
+    then
```
Comment (shovanmaity) [inline hack/build.sh]: can we make separate function which setting CTLNAME and delete old contents of bin .

Comment (sagarkrsd) [reply hack/build.sh]: Right @shovanmaity . My bad !!

**Comment by AmitKumarDas** [comment]
@sagarkrsd can you run the shellcheck in travis builds?

**Comment by sagarkrsd** [comment]
@AmitKumarDas Its already running in travis build since it is included in build of makefile and we are doing make in travis.

File: hack/build.sh [hunk 2]
```diff
@@ -3,42 +3,108 @@
...
-mkdir -p bin/
-
-if [ -z "${CTLNAME}" ];
-then
-    CTLNAME="ndm"
-fi
+deleteOldContents
 
 # If its dev mode, only build for ourself
 if [[ "${M_AGENT_DEV}" ]]; then
```
Comment (pawanpraka1) [inline hack/build.sh]: NDM_AGENT_DEV

Comment (sagarkrsd) [reply hack/build.sh]: Sure @pawanpraka1 . I will change it !

File: Makefile [hunk 3]
```diff
@@ -42,6 +42,13 @@ golint:
 	@gometalinter.v1 --install
 	@gometalinter.v1 --vendor --deadline=600s ./...
 
+# shellcheck target for checking shell scripts linting
+shellcheck: /tmp/shellcheck-latest/shellcheck
```
Comment (pawanpraka1) [inline Makefile]: instead of /tmp/shellcheck-latest/shellcheck have an end point. binshellcheck or getshellcheck?

**Comment by sagarkrsd** [comment]
@pawanpraka1 I had already discussed with @kmova before removing it.

**Pull request merged**